packages/principled-forms/src/form.ts

```diff
@@ -81,7 +81,7 @@ export type Form<T> = Required<
 export const isValid = <T, F extends Form<T>, K extends keyof F>(form: F): boolean =>
   (Object.keys(form) as K[])
     .map(key => form[key] as Field<any>) // `any` b/c TS loses mapped type context here
-    .map(Field.validate)
+    .map(field => Field.validate(field))
```

Comment:
This only exists because I made `validate` take a second boolean argument, and I got type errors because `map` tries to pass the `index` as the second argument.

Reply:
Got it. 🤔 IIRC we had `eager` on the field itself so that e.g. `Email` could just *always* be lazy, right? Any reason not to just go back to that?

Reply:
But I don't think we want `Email` to _always_ be lazy. I'm thinking of the following case.

* I start typing an email: `foo@bar`. With lazy validation, the form is still considered unvalidated.
* I get distracted.
* When I come back, without thinking to finish typing my email, I just tab away to the next field.

At this point, it would be nice to be able to force eager validation.

I think another case where forced eager validation would be nice is the case when the user hits the submit button. At this point, we would want to eagerly validate every field and display error messages accordingly.

Reply:
In short, I think that the eager/lazy distinction is really only useful for text inputs, and I also think it only makes sense the first time a user "visits" a field. After they "leave", we can validate changes.

Reply:
Yeah, that seems right. Good call.